caravel/looper_parser.py

```diff
@@ -56,18 +123,101 @@ def _get_subparser(p):
     return subs[0]
 
 
-def _has_long_opt(act):
-    """ Determine whether the given option defines a long option name. """
-    try:
-        opts = act.option_strings
-    except AttributeError:
-        opts = []
-    for n in opts:
+def _has_long_opt(opt):
+    """
+    Determine whether the given option defines a long option name
+
+    :param list opt: a list of option name(s)
+    :return: bool indicating whether the given option defines a long option name
+
+    """
+    if not isinstance(opt, list):
+        raise TypeError("The opt argument has to be a list of strings.")
+    for n in opt:
         if _is_long_optname(n):
             return True
     return False
 
 
+def _get_long_opt(opt):
+    """
+    Get only the long option name from the option strings
+
+    :param list opt: a list of option name(s)
+    :return: a long option name or empty string
+    """
+    if _has_long_opt(opt):
+        return opt[map(_is_long_optname, opt).index(True)]
```

Comment:
What's `.index(True)` do?

Reply:
finds the index of the list element `== True`

Reply:
Ah got it--sorry, I'm not used to working with lists of bools